<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node">
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="320" />
    <param name="image_height" value="240" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap" />
  </node>
  <node name="trainer" pkg="object_detector" type="trainer_node" output="screen">
    <remap from="image" to="/usb_cam/image_raw" />
  </node>
  <node name="object_detector" pkg="object_detector" type="object_detector_node" output="screen">
    <remap from="image" to="/usb_cam/image_raw" />
    <param name="detection_threshold" value="0.0" />
    <param name="detector_config_file" value="$(find object_detector)/cfg/detector.cfg"/>
  </node>
  <node name="detection_display" pkg="object_detector" type="detection_display_node">
    <remap from="image" to="/usb_cam/image_raw" />
    <remap from="detection" to="/object_detector/detection" />
  </node>
</launch>

